source/slang/core.meta.slang

```diff
@@ -1270,6 +1270,7 @@ $}
 
 //@hidden:
 __magic_type(NullPtrType)
+__intrinsic_type($(kIROp_RawPointerType))
 struct NullPtr
```

Comment:
NullPtr is made to lower as a RawPointerType, because this fixes the SizeOf computation for the nullptr literal in peephole optimization. Previously, it was lowered as an empty struct, so SizeOf returned 0 if computed at the IR level. It used to be computed in `slang-ast-natural-layout.cpp` instead, but that won't work for pointers anymore because we don't know their sizes at the AST level.